### PR TITLE
Refactored Solution - Java Fruit Shop Challenge

### PR DESCRIPTION
**Pull request by kingsedem**
I intend to use this refactored version on my Portfolio, kindly let me know if it is good.
 
 Thanks!

### CODE REVIEW

File: src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java [hunk 1]
```diff
@@ -10,49 +11,43 @@
...
+    Catalog catalog = new Catalog();
+
+    @BeforeEach
+    public void setUpCatalog(){
+        catalog.setPrice("Apples", 4.00);
+        catalog.setPrice("Oranges", 5.50);
+        catalog.setPrice("Bananas", 6.00);
+        catalog.setPrice("Pears", 4.50);
+    }
+    @DisplayName("Should return the current price of a fruit")
```
Comment (globalworming) [inline src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java]: method says `shouldBeAbleToUpdateTheCurrentPriceOfAFruit` but the display name says something different

File: src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java [hunk 2]
```diff
@@ -10,49 +11,43 @@
...
+            assertThat(priceOfBananas).isEqualTo(6.00);
+            assertThat(priceOfPears).isEqualTo(4.50);
         }
 
+    @DisplayName("Should list the names of the currently available fruits in alphabetical order")
+
         @Test
         public void shouldListProductsInAlphabeticalOrder(){
-            Catalog catalog = new Catalog();
-            catalog.setPrice("Cherries", 3.00);
```
Comment (globalworming) [inline src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java]: moved to setup :+1: nice

File: src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java [hunk 3]
```diff
@@ -10,49 +11,43 @@
...
-
-            assertThat(catalog.getProducts()).containsExactly("Apples", "Bananas", "Cherries");
+            assertThat(catalog.getProducts()).containsExactly("Apples", "Bananas", "Oranges", "Pears");
         }
 
 
+    @DisplayName("Should throw a FruitUnavailableException if the fruit is not currently available")
 
         //@Test(expected = FruitUnAvailableException.class)
         @Test
```
Comment (globalworming) [inline src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java]:
```suggestion
        @Test(expected = FruitUnAvailableException.class)

```

we should test that exception gets thrown, right?

**Review by globalworming — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -10,11 +10,8 @@ For example the prices per kilo for the currently available fruits are:
     - Oranges: $5.50
     - Bananas: $6.00
     - Pears:   $4.50
-- You can update the catalog with the current market price of a fruit
-- The Catalog should list the names of the currently available fruit in alphabetical order
-- The Catalog should report the price of a given type of fruit  --- Same as NO.1
+- Update the catalog with the current market price of a fruit
```
Comment (kingsedem) [inline README.md]:
```suggestion
- Update the catalog with the current market price of a fruit.
```

File: src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java [hunk 5]
```diff
@@ -10,49 +11,43 @@
 
 public class TheCatalog {
```
Comment (globalworming) [inline src/test/java/com/serenitydojo/fruitmarket/TheCatalog.java]: by convention the test class should be called "CatalogTest".. or something else with test at the end

**Pull request merged**